packages/fx-core/src/core/FxCore.ts

```diff
@@ -1070,7 +1073,9 @@ export class FxCore implements v3.ICore {
       if (result.isErr()) {
         return err(result.error);
       }
-      config.envInfos[env] = result.value;
+      const envInfo = result.value;
+      envInfo.state = convertEnvStateV3ToV2(envInfo.state) as v3.ResourceStates;
```

Comment:
Why converting to V2 since the method is `getProjectConfigV3`?

Reply:
Ignore the name v3, this method is legacy implementation.

Reply:
This method may already be used with V3 keys, like here: https://github.com/OfficeDev/TeamsFx/blob/dev/packages/vscode-extension/src/debug/commonUtils.ts#L148.

Reply:
Oh, I see.